js/src/components/merchant-center-select-control/index.js

```diff
@@ -2,16 +2,26 @@
  * External dependencies
  */
 import { __, sprintf } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import useExistingGoogleMCAccounts from '.~/hooks/useExistingGoogleMCAccounts';
 import AppSelectControl from '.~/components/app-select-control';
 
-const MerchantCenterSelectControl = ( props ) => {
+/**
+ * @param {Object} props The component props
+ * @param {string} [props.value] The selected value. IF no value is defined, then the first option is selected and onChange function is triggered.
+ * @param {Function} [props.onChange] Callback when the select value changes.
+ * @return {JSX.Element} An enhanced AppSelectControl component.
+ */
+const MerchantCenterSelectControl = ( {
+	value,
+	onChange = () => {},
+	...props
+} ) => {
```

Comment:
Because we are extracting the `value` and the `onChange` params from the `props`, the `AppSelectControl` does not receive the `value` nor `onChange` therefore when you update the dropdown to choose a different account, the MC account is not updated.